### DISCUSSION TITLE
graphql/codegen/plugins/typescript/typescript-mongodb

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#8750] [Docs Discussions]
# graphql/codegen/plugins/typescript/typescript-mongodb

GraphQL Code Generator Documentation

https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-mongodb

**Comment by hoffnung8493** [comment]
for the `@link`, what if its the child documents that have the parent document's `_id` as a foreign key?

**Comment by okanji** [comment]
What if a field in a type is of type Interface? e.g

schema.graphql
```
interface Car {
   id: ID! @id
   make: String! @column
   model: String! @column
}

type Van implements Car {
   d: ID! @id
   make: String! @column
   model: String! @column
   maxLoad: Int!
}

type SportsCar implements Car{
   id: ID! @id
   make: String! @column
   model: String! @column
   topSpeed: Int!
}

// The following type has a field of type Car

type Order {
   id: ID! @id
   customerName: string! @column
   vehicle: Car! @link // This is the field of type interface
}
```

The above is valid graphql schema. Using `@ abstractEntity` on the interface will not work

**Reply by okanji** [reply]
The graphql field resolver will use the resolverType to figure out what `Car` is being uses in the vehicle field. Perhaps we need something similar, unless there is a way to make this work as it is?

**Comment by paul-vd** [comment]
In the [scalars](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript#scalars) config, what is the `ScalarsMap_1`?

**Reply by charpeni** [reply]
I've noticed it as well and created an issue for it: https://github.com/dotansimha/graphql-code-generator/issues/9065.
